test/Microsoft.Health.Fhir.Ingest.Template.UnitTests/CodeValueFhirTemplateFactoryTests.cs

```diff
@@ -280,6 +280,20 @@ public void GivenValidTemplateJsonWithMixedMainAndComponentValues_WhenFactoryCre
                 });
         }
 
+        [Theory]
+        [FileData(@"TestInput/data_CodeValueFhirTemplateInvalid_MissingFields.json")]
+        public void GivenInvalidTemplateJsonMissingFields_WhenFactoryCreate_ThenTemplateCreated_Test(string json)
+        {
+            var templateContainer = JsonConvert.DeserializeObject<TemplateContainer>(json);
+
+            var factory = new CodeValueFhirTemplateFactory();
+
+            var template = factory.Create(templateContainer);
+            Assert.NotNull(template);
+            template.IsValid(out string errors);

```

Comment:
should test the return value here.

Reply:
Updated and based on our discussion, I modified and let the template-type-level factory to throw invalid template exception.